File: backend/app/schemas/tools.py
import uuid
from datetime import date

from pydantic import BaseModel, ConfigDict


class ToolSummary(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    name: str
    slug: str
    description: str
    category: str
    mac_supported: bool
    apple_silicon_supported: bool
    intel_supported: bool
    last_verified_at: date


class ToolDetail(ToolSummary):
    official_url: str
    docs_url: str | None
    install_method: str
    homebrew_command: str | None
    verification_command: str | None
    common_errors: dict | None
    alternatives: list[str] | None


class InstallGuideRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    title: str
    content: dict
    last_verified_at: date


class WizardChecklistItem(BaseModel):
    tool: ToolSummary
    essential: bool
    reason: str

File: frontend/src/types/tools.ts
export interface ToolSummary {
  id: string
  name: string
  slug: string
  description: string
  category: string
  mac_supported: boolean
  apple_silicon_supported: boolean
  intel_supported: boolean
  last_verified_at: string
}

export interface ToolDetail extends ToolSummary {
  official_url: string
  docs_url: string | null
  install_method: string
  homebrew_command: string | null
  verification_command: string | null
  common_errors: Record<string, string> | null
  alternatives: string[] | null
}

export interface WizardChecklistItem {
  tool: ToolSummary
  essential: boolean
  reason: string
}

export type Architecture = 'apple_silicon' | 'intel'
export type Experience = 'beginner' | 'intermediate' | 'advanced'
